src/components/Heading/Heading.tsx

```diff
@@ -1,19 +1,28 @@
 import React, { Component } from 'react';
 import classnames from 'classnames';
+import { AsPropType } from '../../utils/typescript';
 import { HtmlH, HtmlHProps, hLevels } from '../../reset/HtmlH/HtmlH';
 export { hLevels };
 
 export interface HeadingProps extends HtmlHProps {
   /** Change HTML-element */
   variant: hLevels;
   className?: string;
+  asProp?: AsPropType;
 }
 
 const baseClassName = 'fi-heading';
 
 export class Heading extends Component<HeadingProps> {
   render() {
-    const { className, variant = 'h1', as, ...passProps } = this.props;
+    const {
+      className,
+      variant = 'h1',
+      as: asStyled,
+      asProp,
+      ...passProps
+    } = this.props;
+    const as = !!asProp ? asProp : asStyled;
```

Comment:
Do I understand the API correctly if I assume that this checks that only as or AsProp can be defined? AsProp for internal use and as for the public API? I'd like to see this explained / commented somehow.

Reply:
`as` is asProp if defined, if not defined it is AsStyled that is just renamed public API `as`. If both undefined then it is undefined.